src/server.js

```diff
@@ -441,6 +441,44 @@ function initializeClient(socket, client, token, lastMessage) {
 
 	socket.on("sessions:get", sendSessionList);
 
+	if (!Helper.config.public) {
+		socket.on("setting:set", (newSetting) => {
+			if (!newSetting || typeof newSetting !== "object") {
+				return;
+			}
+
+			// Older user configs will not have the clientSettings property.
+			if (!client.config.hasOwnProperty("clientSettings")) {
+				client.config.clientSettings = {};
+			}
+
+			// We do not need to do write operations and emit events if nothing changed.
+			if (client.config.clientSettings[newSetting.name] !== newSetting.value) {
+				client.config.clientSettings[newSetting.name] = newSetting.value;
+
+				// Pass the setting to all clients.
+				client.emit("setting:new", {
+					name: newSetting.name,
+					value: newSetting.value,
+				});
+
+				client.manager.updateUser(client.name, {
+					clientSettings: client.config.clientSettings,
+				});
+			}
+		});
+
+		socket.on("settings:get", () => {
```

Comment:
In terms of socket events, there are both `setting` (including `setting:all`) and `settings` like this guy. Do we want to use `settings` everywhere instead?

Reply:
Yeah that makes sense, initially I had the singular and plural because one was going to pass multiple settings and the other not. That changed along the way so it makes sense to have consistent naming there.